dist/setup/index.js

```diff
@@ -125261,9 +125261,19 @@ function isCacheFeatureAvailable() {
     return false;
 }
 exports.isCacheFeatureAvailable = isCacheFeatureAvailable;
-function getVersionFromFileContent(content, distributionName) {
+function getVersionFromFileContent(content, distributionName, versionFile) {
     var _a, _b, _c, _d, _e;
-    const javaVersionRegExp = /(?<version>(?<=(^|\s|-))(\d+\S*))(\s|$)/;
+    let javaVersionRegExp;
+    if (versionFile == '.tool-versions') {
+        javaVersionRegExp =
+            /^(java\s+)(?:\S*-)?v?(?<version>(\d+)(\.\d+)?(\.\d+)?(\+\d+)?(-ea(\.\d+)?)?)$/m;
```

Comment:
Note that this regex will not accept full java versions commonly generated in `asdf` using `latest` - eg:

```
asdf local java latest:corretto-21
```

...will generate a file with this line:

```
java corretto-21.0.2.13.1
```

The regex doesn't recognise this, so `setup-java` will [fail](https://github.com/guardian/etag-caching/actions/runs/8280948170/job/22658453907#step:3:16) with this message:

>  Error: No supported version was found in file .tool-versions

If we [play around with the regex](https://regex101.com/r/1fsHzU/1), we can see that we have to edit the string down to `java corretto-21.0.2` in order to get it to work:

<img width="1723" alt="image" src="https://github.com/actions/setup-java/assets/52038/8757a553-1ca8-4a37-9b28-0601c58cc1be">